test/unit/offset.js

```diff
@@ -568,4 +568,142 @@ QUnit.test( "iframe scrollTop/Left (see gh-1945)", function( assert ) {
 	}
 } );
 
+(function() {
+	var POSITION_VALUES = [ "static", "relative", "absolute", "fixed" ];
+
+	supportjQuery.each( POSITION_VALUES, function( i, docPosition ) {
+		supportjQuery.each( POSITION_VALUES, function( i, bodyPosition ) {
+			supportjQuery.each( [ "static", "absolute" ], function( i, markerPosition ) {
+
+				testIframe( "coordinates relative to document" +
+							" (html{position:" + docPosition + "} body{position:" + bodyPosition + "} div{position:" + markerPosition + "})",
+						"offset/rel-doc.html", function( assert, $, iframe, doc ) {
+					assert.expect( 256 );
+
+					var docElem = doc.documentElement, // `<html>` element
+						bodyElem = doc.body,
+						marker1 = doc.getElementById( "marker1" ),
+						marker2 = doc.getElementById( "marker2" ),
+						$marker1 = $( marker1 ),
+						$marker2 = $( marker2 ),
+						affectProps, markerTopLeft,
+
+						// Bit as Flag of box-properties
+						FLAG_DOC_MARGIN = 1,
+						FLAG_DOC_BORDER = 2,
+						FLAG_DOC_PADDING = 4,
+						FLAG_BODY_MARGIN = 8,
+						FLAG_BODY_BORDER = 16,
+						FLAG_BODY_PADDING = 32,
```

Comment:
> Why does the code change the `top/left` of `html/body`?
I think that these properties affect `getBoundingClientRect()` method of browser, and `.offset()` method of jQuery calculates with a return value of the `getBoundingClientRect()` without considering `top/left`.
That is, if the result become incorrect by changing `top/left`, it is bug of `getBoundingClientRect()` method (i.e. browser), not jQuery.

If there are such bugs, we need to know so we can work around them (or document our failure to do so).

> Also, `top/left` of `html/body` are usually not changed.

Nor do html or body elements usually have borders. The point is verifying that our functions perform as claimed.

> Why is the code that passes a return value of `.offset()` to another element unnecessary?
…
I think that a cases of an element that has `position:static` are important because `setOffset()` calculates distance from current coordinates. https://github.com/jquery/jquery/issues/3080#issuecomment-215325352

I agree. We should do both `$( "#absolute" ).offset( absoluteOffset )` and `$( "#static" ).offset( absoluteOffset )` and expect the subsequent `.offset()` to match `absoluteOffset`.